src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-revisions-controller.php

```diff
@@ -0,0 +1,484 @@
+					'permission_callback' => array( $this, 'get_item_permissions_check' ),
+					'args'                => $this->get_collection_params(),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Retrieves the query params for collections.
+	 * Taken mostly from WP_REST_Controller->get_collection_params().
+	 *
+	 * @since 6.3.0
+	 *
+	 * @return array Collection parameters.
+	 */
+	public function get_collection_params() {
+		return array(
+			'context'  => array( 'default' => 'view' ),
+			'page'     => array(
+				'description'       => __( 'Current page of the collection.' ),
+				'type'              => 'integer',
+				'default'           => 1,
+				'sanitize_callback' => 'absint',
+				'validate_callback' => 'rest_validate_request_arg',
+				'minimum'           => 1,
+			),
+			'per_page' => array(
+				'description'       => __( 'Maximum number of items to be returned in result set.' ),
+				'type'              => 'integer',
+				'minimum'           => 1,
+				'maximum'           => 100,
+				'sanitize_callback' => 'absint',
+				'validate_callback' => 'rest_validate_request_arg',
+			),
+			'offset'   => array(
+				'description' => __( 'Offset the result set by a specific number of items.' ),
+				'type'        => 'integer',
+			),
+		);
+	}
+
+	/**
+	 * Returns decoded JSON from post content string,
+	 * or a 404 if not found.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string $raw_json Encoded JSON from global styles custom post content.
+	 * @return Array|WP_Error
+	 */
+	private function get_decoded_global_styles_json( $raw_json ) {
+		$decoded_json = json_decode( $raw_json, true );
+
+		if ( is_array( $decoded_json ) && isset( $decoded_json['isGlobalStylesUserThemeJSON'] ) && true === $decoded_json['isGlobalStylesUserThemeJSON'] ) {
+			return $decoded_json;
+		}
+
+		return new WP_Error(
+			'rest_global_styles_not_found',
+			__( 'Cannot find user global styles revisions.' ),
+			array( 'status' => 404 )
+		);
+	}
+
+	/**
+	 * Returns paginated revisions of the given global styles config custom post type.
+	 * The bulk of the body is taken from WP_REST_Revisions_Controller->get_items,
+	 * but global styles does not require as many parameters.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param WP_REST_Request $request The request instance.
+	 * @return WP_REST_Response|WP_Error
+	 */
+	public function get_items( $request ) {
+		$parent = $this->get_parent( $request['parent'] );
+
+		if ( is_wp_error( $parent ) ) {
+			return $parent;
+		}
+
+		$global_styles_config = $this->get_decoded_global_styles_json( $parent->post_content );
+
+		if ( is_wp_error( $global_styles_config ) ) {
+			return $global_styles_config;
+		}
+
+		if ( wp_revisions_enabled( $parent ) ) {
+			$registered = $this->get_collection_params();
+			$query_args = array(
+				'post_parent'      => $parent->ID,
+				'post_type'        => 'revision',
+				'post_status'      => 'inherit',
+				'posts_per_page'   => -1,
```

Comment:
-1 calls need be avoided at ALL COSTS. There is no case where this should be done in production. Please remove this line.

Reply:
Nevermind this was in the revisioins controller. Odd.

Reply:
Oh, could you explain why? 

Should it not also be here as well?

https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/rest-api/endpoints/class-wp-rest-revisions-controller.php#L254

Reply:
It should not no. Unlimited queries can take database servers down if you are not careful.

Reply:
All good! Thanks for raising it anyway. I pinched it from the revisions controller.